File: 0342.4的幂/0342-4的幂.rs
impl Solution {
    pub fn is_power_of_four(n: i32) -> bool {
        if n == 0 {
            return false;
        } else if n == 1{
            return true;
        } else if n % 4 != 0 {
            return false;
        } else {
            return Self::is_power_of_four(n/4);
        }
    }
}

File: 0342.4的幂/0342-4的幂.py
class Solution:
    def isPowerOfFour(self, num: int) -> bool:
        def h(n):
            if n == 1:
                return True
            if n % 4 != 0 or n < 1:
                return False
            return h(n // 4)

        return h(num)
